### ISSUE TITLE
Questions about head model recovery

### ISSUE THREAD

**Issue by bearprin** [GAP-LAB-CUHK-SZ/HairStep#18]
Hi

Thanks for your impressive work on single-view hair reconstruction.

I noticed that your head model recovered from the single view is extremely good-looking, which has the shoulder and the same neck length as the input image.

To my knowledge, useful methods such as FLAME or PIXIE cannot yield the same results.

Could you let me know which method you are using to get the high-fidelity head model?

Thanks for your help!

**Comment by Zvyozdo4ka** [comment]
It takes the length as the height of body from top of the head till the edge of photo. 
![image](https://github.com/GAP-LAB-CUHK-SZ/HairStep/assets/74532816/2352c5dc-f93b-4e21-98fc-92621175193f)

Any ideas how to fix it?

@bearprin Have you worked in FLAME?

**Comment by Zvyozdo4ka** [comment]
I found this:

```
        # NOTE: for newer PyTorch (1.3~), it seems that training results are degraded due to implementation diff in F.grid_sample
        # if the pretrained model behaves weirdly, switch with the commented line.
        # NOTE: I also found that "bicubic" works better.
        
        # up2 = F.interpolate(low3, scale_factor=2, mode='bicubic', align_corners=True)
        up2 = F.interpolate(low3, size=(up1.shape[2],up1.shape[3]), mode='bicubic', align_corners=True)
        up2 = F.interpolate(low3, scale_factor=2, mode='nearest')
```

I tried all this variations, and it is still weird, with long hair for short male hairstyles.

@bearprin Have you tried to check your PyTorch?

Mine is `1.9.0+cu111` as it required.

**Comment by bearprin** [comment]
@Zvyozdo4ka 

Hi 

I think you may use the same types of images as the dataset that captures people in front with a clear background such as the wall.

FLAME works but it cannot produce the same pics with details and shoulder as the paper showed. Also, the coordinate system is a little different.

**Comment by Zvyozdo4ka** [comment]
@bearprin 

> 
> I think you may use the same types of images as the dataset that captures people in front with a clear background such as the wall.

I did as you suggested, but it did not work on my images, and image from internet, but worked on image from paper:

![image](https://github.com/GAP-LAB-CUHK-SZ/HairStep/assets/74532816/4483551e-dfa8-4408-81e6-a78a98ea97ec)

that is a paradox for me

> 
> FLAME works but it cannot produce the same pics with details and shoulder as the paper showed. Also, the coordinate system is a little different.

Could you share some tests from Flame?

**Comment by Zvyozdo4ka** [comment]
@bearprin which FLAME version did you use?

**Comment by bearprin** [comment]
@Zvyozdo4ka 

My env is 1.31 pytorch + cu11

I use [PIXLE](https://github.com/yfeng95/PIXIE) to recover face

Btw, I think you may try the photo with long hair

**Comment by Zvyozdo4ka** [comment]
> My env is 1.31 pytorch + cu11
@bearprin  How did you set such environment? I tried several times, but always it is written that 1.3.1 is incompatible with cuda11. Please share your tips about your env settings.
 
> I use [PIXLE](https://github.com/yfeng95/PIXIE) to recover face

Thank you. I shall try it.


> Btw, I think you may try the photo with long hair
But i need to solve it for short hair. For long hair it works almost fine, except that on the top of the head their is a bold area:
![image](https://github.com/GAP-LAB-CUHK-SZ/HairStep/assets/74532816/68f51985-e61f-4e46-b2ee-3e81a0d61976)

**Comment by bearprin** [comment]
@Zvyozdo4ka 
Sorry for my mistakes. I actually use 1.9.0 torch + cuda11.1

In my opinion, it is difficult to recover the short hair by existing methods.

On the one hand, short hair is very challenging

On the other hand, most hair in the existing dataset is long hair rather than short

**Comment by Zvyozdo4ka** [comment]
> @Zvyozdo4ka Sorry for my mistakes. I actually use 1.9.0 torch + cuda11.1

@bearprin Thank you for clarifying! It was really confusing how you did installed old torch on cuda 11.1

> In my opinion, it is difficult to recover the short hair by existing methods.
> 
> On the one hand, short hair is very challenging

yes, know, but really need to solve this problem. I think that's also due to imbalanced dataset, as they mentioned in dataset 80% are female images and 20% are male images. Also they collected 300 Short hair, 300 Middle hair and 650 Long hair.

**Comment by PAULYZHENG** [comment]
For tackling the problem of short hair, please refer #10 . You can train a image-to-image translation module to predict the back mask from the frontal stand map, then using this mask to cut the redundant hair by projection.

Later I may release this strategy, but won't be very soon.

**Comment by Zvyozdo4ka** [comment]
> For tackling the problem of short hair, please refer #10 . You can train a image-to-image translation module to predict the back mask from the frontal stand map, then using this mask to cut the redundant hair by projection.
> 
> Later I may release this strategy, but won't be very soon.

Hello. Have you released this strategy by any chance?
